InterfaceFactory/CommHandler4Modbus/src/main/java/communicator/common/helper/DeviceDescriptionLoader.java

```diff
@@ -185,20 +186,75 @@ private C createResource(String resourcePath, String deviceDescXml, Properties p
 
 		Resource resource = domain.createResource(resourcePath);
 
+		// load the XML for the first time, in order to get configuration list
+		InputStream is = null;
+		try {
```

Comment:
There is a nice 'try with resource' feature in Java. You could replace the try/catch/finally block with:

```java
		try (InputStream is = IOUtils.toInputStream(deviceDescXml, StandardCharsets.UTF_8)) {
			resource.load(is, null);
		}
```

Reply:
Yes, this seems a better approach